docs/pages/example/snapshot.html

```diff
@@ -0,0 +1,120 @@
+<style>
+  html, body {
+    margin: 0;
+  }
+
+  #map {
+    position: absolute;
+    width:50%;
+    height:100%;
+    background: rgba(0,255,0,0.5);
+  }
+
+  #snapshot {
+    position: absolute;
+    left:50%;
+    background:rgba(0,0,255,0.5);
+    width:50%;
+    height:100%;
+  }
+
+  h1 {
+    color:white;
+    text-align:center;
+    position: relative;
+    top:50%;
+    transform: translateY(-50%);
+    margin:0;
+  }
+ 
+</style>
+
+<div id='map' class='window'></div>
+
+<canvas id='snapshot' class='window'></div>
+
+<script>
+
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/dark-v9',
+    center: {
+        lat: 37.4036,
+        lng: -122.0325
+    },
+    zoom: 10,
+    preserveDrawingBuffer: true// required for the map's canvas to be exported to a PNG
+});
+
+var snapshot = document.querySelector('#snapshot');
+var ctx = snapshot.getContext('2d');
+
+// draw snapshot canvas 
+window.addEventListener('load', function() {
+    var text = 'Click on the map';
+    snapshot.height = window.innerHeight;
+    snapshot.width = window.innerWidth / 2;
+    ctx.fillStyle = 'white';
+    ctx.textAlign = 'center';
+    ctx.textBaseline = 'middle';
+    ctx.font = Math.round(snapshot.width / 10) + 'px Arial';
+    ctx.fillText(text, snapshot.width / 2, snapshot.height / 2);
+}, false);
+
+map.on('click', function() {
+    var png = map.getCanvas().toDataURL();
+    var copy = new Image();
+    copy.src = png;
+    copy.onload = function() {
+        ctx.drawImage(copy, 0, 0);
+        logo();
+        textbox();
+    };
+});
+
+function textbox() {
+    // Please attribute mapbox if using non-commercially
+    // https://www.mapbox.com/help/how-attribution-works
+    var text = '© Mapbox © OpenStreetMap';
+    if (snapshot.width < 300) text = '© OSM';
+    var fontsize = 14;
+    var height = fontsize + 6;
+    var width = (text.length + 3) / fontsize * 100;
+    ctx.fillStyle = 'rgba(255,255,255,0.4)';
+    ctx.fillRect(snapshot.width - width, snapshot.height - height, width, height);
+    ctx.font = fontsize + 'px Arial';
+    ctx.fillStyle = 'black';
+    ctx.textAlign = 'left';
+    ctx.textBaseline = 'bottom';
+    ctx.fillText(text, snapshot.width - width + 3, snapshot.height - 3);
+}
+
+function logo() {
+    // Please attribute mapbox if using non-commercially
+    // https://www.mapbox.com/help/how-attribution-works
+    var img = new Image();
+    var logo = mapboxlogo();
+    img.src = logo.image64;
+    img.onload = function() {
+    //(!) Doesn't work in Firefox
+        ctx.drawImage(img, 3, snapshot.height - logo.height - 4, logo.width, logo.height);
+    };
+}
+
+function mapboxlogo() {
+
+    // logo image
+    var a = document.querySelector('a.mapboxgl-ctrl-logo');
+    var style = window.getComputedStyle(a, false);
+    var base64 = style.backgroundImage.slice(31, -2);
+    var b64Start = 'data:image/svg+xml;base64,';
+    var image64 = b64Start + base64;
```

Comment:
we're not actually using base64 svg anymore since #6513, so I think you'll want to use `"data:image/svg+xml;charset=utf-8,"`

Reply:
I think I am missing something. When I console.log the backgroundImage it seems to indicate it is a base64.

https://github.com/samsammurphy/mapbox-gl-js/blob/de8d14bc7d6c544b47950f68aa60ac3e2990e721/docs/pages/example/snapshot.html#L137

Reply:
simplified accordingly, just using dataURLs as standard approach (with firefox bugfix tacked on later)